### PR TITLE
Add support to paginate as EditOriginalResponse

### PR DESCRIPTION
**Pull request by bakabun**
# Summary
Add support for `SendPaginatedResponseAsync` to send pagination as `EditOriginalResponse`

# Details
Currently there's no proper way to send paginated message after sending `DeferredChannelMessageWithSource`. Because you're supposed to "edit the response" after you're sending defer signal.

Another alternative to "editing the response" is by sending paginated message as follow up message. But I doubt it's standard way to send response after sending defer signal.

# Changes proposed
Add optional parameter `asEditResponse` to `SendPaginatedResponseAsync`. Default to `false`.

# Notes
Idk if there's better approach for this problem so any suggestions will be welcomed.

### CODE REVIEW

File: DSharpPlus.Interactivity/InteractivityExtension.cs [hunk 1]
```diff
@@ -753,13 +754,25 @@ public async Task SendPaginatedResponseAsync(DiscordInteraction interaction, boo
...
+            else
+            {
+                var builder = new DiscordInteractionResponseBuilder()
+                    .WithContent(pages.First().Content)
+                    .AddEmbed(pages.First().Embed)
+                    .AsEphemeral(ephemeral)
+                    .AddComponents(bts.ButtonArray);
+
+                await interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);
+            }
```
Comment (VelvetThePanda) [inline DSharpPlus.Interactivity/InteractivityExtension.cs:775]: Nitpick: Duplicated code. You can put the response in the if/else.

File: DSharpPlus.Interactivity/InteractivityExtension.cs [hunk 2]
```diff
@@ -753,13 +754,25 @@ public async Task SendPaginatedResponseAsync(DiscordInteraction interaction, boo
...
-                .AddComponents(bts.ButtonArray);
+            if (asEditResponse)
+            {
+                var builder = new DiscordWebhookBuilder()
+                    .WithContent(pages.First().Content)
+                    .AddEmbed(pages.First().Embed)
+                    .AddComponents(bts.ButtonArray);
 
-            await interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);
+                await interaction.EditOriginalResponseAsync(builder);
```
Comment (VelvetThePanda) [inline DSharpPlus.Interactivity/InteractivityExtension.cs:764]: If I'm not mistaken, this returns `Task<DiscordMessage>` so what you can do is actually something along the lines of

```cs
DiscordMessage message = null;

if (asEditResponse)
    message = await interaction.EditOriginalResponseAsync(builder);
else 
    await interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);

message ??= await interaction.GetOriginalResponseAsync();
```

**Review by VelvetThePanda — CHANGES_REQUESTED** [review]
Could use some cleanup, but looks good to me.

File: DSharpPlus.Interactivity/InteractivityExtension.cs (hunk 2 above)
Comment (bakabun) [reply DSharpPlus.Interactivity/InteractivityExtension.cs:764]: `EditOriginalResponseAsync` and `CreateResponseAsync` uses different builder and `DiscordWebhookBuilder` can't be casted to `DiscordInteractionResponseBuilder` and vice versa so merging both builder seems not possible.

**Review by IDoEverything — APPROVED** [review]

**Pull request merged**